icu4c/source/i18n/ulocdata.cpp

```diff
@@ -288,6 +291,11 @@ ulocdata_getLocaleDisplayPattern(ULocaleData *uld,
     if (U_FAILURE(*status))
         return 0;
 
+    if (uld->langBundle == nullptr) {
+        *status = U_INVALID_STATE_ERROR;
```

Comment:
I think it would be more appropriate to set a U_MISSING_RESOURCE_ERROR, assuming that that's why the ures_open() call failed.

It would be even cleaner to store that ures_open() call's UErrorCode in a new field in "uld" but maybe that's overkill.

Reply:
I made it catch only U_MISSING_RESOURCE_ERROR and then we know we can return U_MISSING_RESOURCE_ERROR because it's the only way to get to the state of a null langBundle.